src/diff/index.js

```diff
@@ -101,10 +109,13 @@ export function diff(parentDom, newVNode, oldVNode, context, isSvg, excessDomChi
 				if (c.componentWillUpdate!=null) {
 					c.componentWillUpdate(newProps, c._nextState, cctx);
 				}
-			}
 
-			oldProps = c.props;
-			oldState = c.state;
+				if (c.componentDidUpdate!=null) {
+					c._renderCallbacks.push(() => {
+						c.componentDidUpdate(oldProps, oldState, snapshot);
```

Comment:
Note, I've moved the queuing of `cDU` up here to reuse the `isNew` check and to also no longer need to check `oldProps == null` as a way to determine if `sCU` returned `false`. Plus it lives with the rest of the lifecycle methods so I think reasoning about lifecycle methods will be a little easier.